tls/s2n_connection.c

```diff
@@ -1097,7 +1097,11 @@ int s2n_connection_get_client_hello_version(struct s2n_connection *conn)
 {
     POSIX_ENSURE_REF(conn);
 
-    return conn->client_hello_version;
```

Comment:
How confident are you in our test coverage of s2n_connection_get_client_hello_version? I'm not asking because I see anything wrong, I'm more worried that we have no tests and we might have a behavior change for this API.